frontend/leaderboard/src/Challenges.re

```diff
@@ -1,172 +1,61 @@
 module StringMap = Map.Make(String);
 
-// Helper functions for gathering metrics
-let printMap = map => {
-  StringMap.mapi(
-    (key, value) => {
-      Js.log(key);
-      Js.log(value);
-    },
-    map,
-  );
-};
-
-let calculateProperty = (f, blocks) => {
-  blocks
-  |> Array.fold_left((map, block) => {f(map, block)}, StringMap.empty);
-};
-
-let incrementMapValue = (key, map) => {
-  map
-  |> StringMap.update(key, value => {
-       switch (value) {
-       | Some(valueCount) => Some(valueCount + 1)
-       | None => Some(1)
-       }
-     });
-};
-
-// Gather metrics
-let getBlocksCreatedByUser = blocks => {
-  blocks
-  |> Array.fold_left(
-       (map, block: Types.NewBlock.data) => {
-         incrementMapValue(block.creatorAccount.publicKey, map)
-       },
-       StringMap.empty,
-     );
-};
-
-let calculateTransactionSent = (map, block: Types.NewBlock.data) => {
-  block.transactions.userCommands
-  |> Array.fold_left(
-       (transactionMap, userCommand: Types.NewBlock.userCommands) => {
-         incrementMapValue(userCommand.fromAccount.publicKey, transactionMap)
-       },
-       map,
-     );
-};
-
-let getTransactionSentByUser = blocks => {
-  blocks |> calculateProperty(calculateTransactionSent);
-};
-
-let calculateSnarkWorkCount = (map, block: Types.NewBlock.data) => {
-  block.snarkJobs
-  |> Array.fold_left(
-       (snarkMap, snarkJob: Types.NewBlock.snarkJobs) => {
-         incrementMapValue(snarkJob.prover, snarkMap)
-       },
-       map,
-     );
-};
-
-let getSnarkWorkCreatedByUser = blocks => {
-  blocks |> calculateProperty(calculateSnarkWorkCount);
-};
-
-let getSnarkFeesCollected = blocks => {
-  Array.fold_left(
-    (map, block: Types.NewBlock.data) => {
-      Array.fold_left(
-        (map, snarkJob: Types.NewBlock.snarkJobs) => {
-          StringMap.update(
-            snarkJob.prover,
-            feeCount =>
-              switch (feeCount) {
-              | Some(feeCount) => Some(Int64.add(feeCount, snarkJob.fee))
-              | None => Some(snarkJob.fee)
-              },
-            map,
-          )
-        },
-        map,
-        block.snarkJobs,
-      )
+let addPointsToUsersWithAtleastN =
+    (getMetricValue, threshold, pointsToReward, metricsMap) => {
+  StringMap.fold(
+    (key, metric, map) => {
+      switch (getMetricValue(metric)) {
+      | Some(metricValue) =>
+        metricValue >= threshold
+          ? StringMap.add(key, pointsToReward, map)
+          : StringMap.add(key, 0, map)
```

Comment:
I think we can just return map here, right?

Reply:
yes, absolutely. made the fix.